src/Notes.cc

```diff
@@ -108,8 +110,8 @@ void
 Note::dump(StoreEntry *entry, const char *k)
 {
     for (const auto &v: values) {
-        storeAppendPrintf(entry, "%s %.*s %s",
-                          k, key().length(), key().rawContent(), ConfigParser::QuoteString(SBufToString(v->value())));
+        storeAppendPrintf(entry, "%s " SQUIDSBUFPH " %s",
+                          k, SQUIDSBUFPRINT(key()), ConfigParser::QuoteString(SBufToString(v->value())));
```

Comment:
Please use `PackableStream os(*entry)` and avoid preserving storeAppendPrintf() (among other long-term advantages). This is especially relevant in a PR that focuses on removing appendf() calls!

P.S. You do _not_ have to convert dump_acl_list() below to use a stream here.